src/main/java/org/jitsi/jigasi/xmpp/CallControl.java

```diff
@@ -146,99 +159,91 @@ public CallControl(ConfigurationService config)
+        String from = iq.getSource();
+        String to = iq.getDestination();
 
-                ctx.setDestination(to);
+        ctx.setDestination(to);
 
-                String roomName = dialIq.getHeader(ROOM_NAME_HEADER);
-                ctx.setRoomName(roomName);
+        String roomName = iq.getHeader(ROOM_NAME_HEADER);
+        ctx.setRoomName(roomName);
 
-                String roomPassword = dialIq.getHeader(ROOM_PASSWORD_HEADER);
-                ctx.setRoomPassword(roomPassword);
-                if (roomName == null)
-                    throw new RuntimeException("No JvbRoomName header found");
+        String roomPassword = iq.getHeader(ROOM_PASSWORD_HEADER);
+        ctx.setRoomPassword(roomPassword);
+        if (roomName == null)
+            throw new RuntimeException("No JvbRoomName header found");
 
-                logger.info(
-                    "Got dial request " + from + " -> " + to
-                        + " room: " + roomName);
+        logger.info(
+            "Got dial request " + from + " -> " + to + " room: " + roomName);
 
-                String callResource = ctx.getCallResource();
-
-                if(TRANSCRIPTION_DIAL_IQ_DESTINATION.equals(to))
-                {
-                    transcriptionGateway.createOutgoingCall(ctx);
-                }
-                else
-                {
-                    sipGateway.createOutgoingCall(ctx);
-                }
-
-                callResource = "xmpp:" + callResource;
+        AbstractGatewaySession session;
+        if(TRANSCRIPTION_DIAL_IQ_DESTINATION.equals(to))
+        {
+            session = transcriptionGateway.createOutgoingCall(ctx);
+        }
+        else
+        {
+            session = sipGateway.createOutgoingCall(ctx);
+        }
 
-                return RefIq.createResult(iq, callResource);
-            }
-            else if (iq instanceof HangUp)
-            {
-                // FIXME: 23/07/17 Transcription sessions should also be able
-                // to be hangup ?
+        if (createdSession != null && createdSession.length == 1)
+        {
+            createdSession[0] = session;
+        }
 
-                HangUp hangUp = (HangUp) iq;
+        return RefIq.createResult(iq, "xmpp:" + ctx.getCallResource());
+    }
 
-                String callResource = hangUp.getTo();
+    /**
+     * Handles an {@link IQ} stanza of type {@link IQ.Type#set} which
+     * represents a request.
+     *
+     * @param iq the {@link IQ} of type {@link IQ.Type#set} which represents
+     * the request to handle
+     * @return an {@link IQ} stanza which represents the response to the
+     * specified request.
+     */
+    public IQ handleHangUp(HangUp iq)
+        throws CallControlAuthorizationException
+    {
+        checkAuthorized(iq);
 
-                SipGatewaySession session = sipGateway.getSession(callResource);
+        // FIXME: 23/07/17 Transcription sessions should also be able
```

Comment:
anything here that needs to be addressed before merging?

Reply:
This must be a mismatched merge, the note isn't from me.

Reply:
Actually not a mismatched merge. The note was there before, just at a different position (in the if/multi-else type check of handleIQ).